Add owner/group/mode parameter for libvirt_pool

baurmatt opened the pull request:
This commit adds support for managing the target permissions by
extending the libvirt_pool type with the target_(owner|group|mode)
parameters.

Fixes #13.

trefzer commented on lib/puppet/provider/libvirt_pool/virsh.rb:
```diff
@@ -174,10 +174,32 @@ def buildPoolXML(resource)
+          targetPermissionsOwnerEl = targetPermissionsEl.add_element 'owner'
+	  targetPermissionsOwnerEl.add_text targetOwner.to_s
+	end
+
```
this is not a correct alignment ...

trefzer commented on lib/puppet/provider/libvirt_pool/virsh.rb:
```diff
@@ -174,10 +174,32 @@ def buildPoolXML(resource)
+          targetPermissionsGroupEl = targetPermissionsEl.add_element 'group'
+	  targetPermissionsGroupEl.add_text targetGroup.to_s
+	end
+
```
dito

trefzer commented on lib/puppet/provider/libvirt_pool/virsh.rb:
```diff
@@ -174,10 +174,32 @@ def buildPoolXML(resource)
+        if targetMode
+          targetPermissionsModeEl = targetPermissionsEl.add_element 'mode'
+	  targetPermissionsModeEl.add_text targetMode.to_s
+	end
```
dito

baurmatt commented:
It was easier then i thought oO

- Fixed alignments
- Added the `puppet strings` generate documentation of libvirt_pool to README.md
- Added a warning if target_(owner|group|mode) is set put `type` isn't fs/dir

trefzer commented:
Thanks !
